### DISCUSSION TITLE
setuptools >= 69.3

### DISCUSSION THREAD

**Discussion by ftapajos** [sqlalchemy/sqlalchemy#11818] [Ideas]
[Some distributions](http://autobuild.buildroot.org/results/c531b6e63c5121138e8e8a1c5d1501d9d7e1391e/build-end.log) already have updated setuptools, but commit 2607262110bdc5c5dc96fc19ddca895a15a58e4e pinned it to < 69.3.
These distributions will have to pin sqlalchemy to version 2.0.31 until it is compatible with at least version 71.0.3. Is there any safe workaround that may be implemented when downgrading setuptools is not an option?

**Comment by CaselIT** [comment]
Hi,

There should be no issue wrt using a more modern setuptools version, if I remember correctly it using a more modern version would break some distribution since it changes the file names.
It's probably fine to patch that version constraint though

But I'm not sure why is it a problem. Why are distribution going against the python build tools, that have converged into using isolated build environment and instead use a global one? Using the mechanism specified by pep517 etc there shouldn't be any issue like these.

@zzzeek do you remember what distro had issue that required the pin?

**Reply by zzzeek** [reply]
it likely broke my own build scripts first, which is not a big deal I can fix those, but then changing our filename mid 2.0.x series seemed like a recipe for disaster.

sure we can just flip the switch and go with it, why not

**Comment by zzzeek** [comment]
https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5478
